docs/api/api.yml

```diff
@@ -637,7 +639,10 @@ components:
         not exist, either.
 
     ConflictZipMustBeOnlyFile:
-      description: A multipart/form-data message contained a ZIP part but also other parts.
+      description: Conflict. A multipart/form-data message contained a ZIP part but also other parts.
+
+    ConflictZipOverwritesFilesOnServer:
+      description: Conflict. Unpacked ZIP file would overwrite existing files on the server.
```

Comment:
I understand the potential problem. The best solution would be to put the conflict check and the move in a synchronized block. The last to finish will then get the conflict status code.

Reply:
**Is overwrite still allowed in case multiple non-zip files are uploaded?**

If yes then we can implement the lock and check inside `copyPlainItemsTo`. If no it gets more complicated.

Current logic:
https://github.com/DANS-KNAW/easy-deposit-api/blob/2c1a34e02671a6762f0e04d96ce3b7f393eaecc2/src/main/scala/nl.knaw.dans.easy.deposit/servlets/DepositServlet.scala#L139-L142

By the way, I was mistaken about the intended worker thread: that applied to the submit action. Submit is supposed to validate the bag in the main thread but copy the complete deposit (and recalculate check sums) via the staging area to `easy-ingest-flow-inbox`.

Reply:
Note: if the answer above is yes, lock the root of the data directory in the bag, not the target `dir_path`. It prevents concurrent uploads to different levels of an existing tree.

Reply:
@jo-pol : will a simple synchronized block not be sufficient?